mautic/mautic
# Reply API endpoint

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Automated tests included? | Y
| Related user documentation PR URL | /
| Related developer documentation PR URL | https://github.com/mautic/developer-documentation/pull/125
| Issues addressed (#s or URLs) | /
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This PR adds new `POST api/emails/reply/{trackingHash}` API endpoint that can create a record about an email reply to some specific email stat that is identified by the `trackingHash`.

#### Steps to test this PR:
1. Execute `POST api/emails/reply/{trackingHash}` while authenticated to Mautic API and use a tracking hash from one of the records in the `email_stats` table.
2. It should create a new record in the `email_stat_replies` table and mark the `email_stats` record as read if not already.
3. The `GET api/stats` endpoint now contains also `email_stat_replies` table.
